modules/data-entry/src/main/frontend/src/dataHomepage/SubjectTypes.jsx

```diff
@@ -16,45 +16,102 @@
 //  specific language governing permissions and limitations
 //  under the License.
 //
-import React from "react";
+import React, { useState, useEffect } from "react";
 import LiveTable from "./LiveTable.jsx";
-import SubjectType from "../questionnaire/SubjectType.jsx";
 
-import { Button, Card, CardContent, CardHeader, withStyles } from "@material-ui/core";
+import { Button, Card, CardContent, CardHeader, IconButton, Tooltip, withStyles } from "@material-ui/core";
 import QuestionnaireStyle from "../questionnaire/QuestionnaireStyle.jsx";
+import SubjectTypeDialog from "../questionnaire/SubjectTypeDialog.jsx";
+import NewItemButton from "../components/NewItemButton.jsx";
 import DeleteButton from "./DeleteButton.jsx";
+import EditIcon from "@material-ui/icons/Edit";
+import AddIcon from "@material-ui/icons/Add";
+
+// Get a flat list of subject type parents as labels separated by " / "
+function getTextHierarchy (path, subjects) {
+  let names = path.replace("/SubjectTypes/", "").split("/");
+  let hierarchy = "";
+  for (let name of names) {
+    let subject = subjects.find( item => item["@name"] === name );
+    hierarchy = ( hierarchy ? hierarchy + " / " : "") + (subject?.label || name);
+  }
+  return hierarchy;
+}
+
+function EditSubjectTypeButton(props) {
+  const { classes, onClick } = props;
+  return(
+    <Tooltip title={"Edit Subject Type"}>
+      <IconButton className={classes.actionButton} onClick={onClick}>
+        <EditIcon />
+      </IconButton>
+    </Tooltip>
+  )
+}
 
 function SubjectTypes(props) {
   const { classes } = props;
-
+  const [ newSubjectTypePopperOpen, setNewSubjectTypePopperOpen ] = useState(false);
+  const [ updateData, setUpdateData ] = useState(false);
+  const [ subjectData, setSubjectData ] = useState([]);
+  const [ editSubject, setEditSubject ] = useState(null);
+  const [ isEdit, setIsEdit ] = useState(false);
   const columns = [
     {
-      "key": "label",
-      "label": "Label",
-      "format": "string",
-      "link": "dashboard+field:@path",
+      "key": "",
+      "label": "Subject Type",
+      "format": (row) => (getTextHierarchy(row['@path'], subjectData)),
     },
     {
-      "key": "jcr:created",
-      "label": "Created on",
-      "format": "date:YYYY-MM-DD HH:mm",
+      "key": "",
+      "label": "Subjects",
+      "format": (row) => (<a href={"/content.html/Subjects#" + row['@name']} title={"Show subjects of type " + row.label} target="_blank">{row.subjectsNumber || 0}</a>),
```

Comment:
We no longer need to open it in a separate tab
```suggestion
      "format": (row) => (<a href={"/content.html/Subjects#" + row['@name']} title={"Show subjects of type " + row.label}>{row.subjectsNumber || 0}</a>),
```